### PR TITLE
[JENKINS-34522] Expose the noun for a item as a task via AlternativeUiTextProvider

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-34522](https://issues.jenkins-ci.org/browse/JENKINS-34522)

@reviewbybees

### CODE REVIEW

File: core/src/main/resources/hudson/views/BuildButtonColumn/column.properties [hunk 1]
```diff
@@ -20,6 +20,6 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-Build_scheduled=Build scheduled
-Schedule_a_build=Schedule a build for {0}
-Schedule_a_build_with_parameters=Schedule a build with parameters for {0}
+Task_scheduled={0} scheduled
+Schedule_a_task=Schedule a {1} for {0}
+Schedule_a_task_with_parameters=Schedule a {1} with parameters for {0}
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column.properties]: At least delete the now obsolete entries from the other localizations.

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/resources/hudson/views/BuildButtonColumn/column.properties (hunk 1 above)
Comment (jglick) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column.properties]: BTW `core/move-l10n.groovy` is the easiest way when keeping other keys.

**Review by jglick — APPROVED** [review]
Fine subject to existence of downstream PR (presumably in `branch-api`?) demonstrating consumption of the API.

**Comment by stephenc** [comment]
@jenkinsci/code-reviewers

**Review by oleg-nenashev — DISMISSED** [review]
+1 for using `core/move-l10n.groovy` or whatever to keep the existing translations in the case when the task is actually a build.

**Comment by stephenc** [comment]
@oleg how can we keep existing and template them at the same time?

**Comment by daniel-beck** [comment]
AFAIU @jglick is just referring to the `--delete` mode that removes obsolete entries without deleting files. These entries are gone, period. No way we can reasonably update them.

**Comment by stephenc** [comment]
@daniel-beck correct and as the file only had three entries and all three are gone, it did not make sense to retain files just to keep copyright headers

**Comment by oleg-nenashev** [comment]
@daniel-beck @stephenc Actually you could retain them. By just adding a logic to `BuildColumn#taskNoun()`, which would check instance of the object and return the legacy translation as a fallback for `AbstractProject`s, where the original translation is valid. 

Otherwise I feel incomfortable that we just screw up translations just for delivering shiny features in particular plugins. And with the current translation policy we will never get the 95% of translations back. I would not say I care much, but the "screw up them and go forward" policy stinks bad in regards of the existing translation contributors.

**Comment by daniel-beck** [comment]
@oleg-nenashev It seems unnecessary to make the code more complicated to save a minute of translating strings per language, if that. Out of necessity we already have often very complicated code -- no need to add more of that if we don't need to. You're saying you're thinking of the *contributors*, but does that include those trying to understand the code? If the *only* reason is the existing translations, rather than localizability more generally, I don't understand your argument at all.

I did https://github.com/jenkinsci/jenkins/pull/2777, consisting of about 500 strings, in several hours hours (didn't count, and much time was wasted due to #2779, I don't count that). I expect to spend another 1-2 hours on this. Another one or two strings wouldn't even be noticed.

That language went from 1289 strings in 1.580 to 1203 strings in 2.49, and most of those (65 out of 86) were lost between 1.656 and 2.0, indicating it was due to how the "agent" terminology change was implemented (probably should have been done differently in hindsight). Otherwise, we don't lose existing translations anywhere near a relevant pace sufficient to make a big deal out of this.

File: core/src/main/resources/hudson/views/BuildButtonColumn/column_pl.properties [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-# THE SOFTWARE.
-
-Build_scheduled=Zadanie zosta\u0142o zaplanowane
-Schedule_a_build=Dodaj zadanie {0} do kolejki
-Schedule_a_build_with_parameters=Uruchom z parametrami dla {0}
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column_pl.properties]: @damianszczepanik There's a discussion currently going on in #2772 about the impact of that change on localizations. I'd like to know, would something like this, i.e. no effort done on retaining existing translations, annoy or upset you? Or would you consider it just part of ongoing development?

**Comment by stephenc** [comment]
Nobody else has expressed an opinion... merging

File: core/src/main/resources/hudson/views/BuildButtonColumn/column_pl.properties (hunk 2 above)
Comment (damianszczepanik) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column_pl.properties]: Sorry for delay, will take a look at this

**Pull request merged**